provider/azureProvider.js

```diff
@@ -178,7 +179,24 @@ return new BbPromise((resolve, reject) => {
       }
     }
 
-    const template = JSON.parse(fs.readFileSync(templateFilePath, 'utf8'));
+    let template = JSON.parse(fs.readFileSync(templateFilePath, 'utf8'));
+
```

Comment:
The current implementation makes the environment variable configuration for the Azure provider behave like it behaves with the other Serverless providers. I think it is valuable to align the behavior between providers wherever possible. If you don't want to set some variable, then you don't define it in your `serverless.yml` file.